### DISCUSSION TITLE
Use community toolkit inside MauiClassLibrary

### DISCUSSION THREAD

**Discussion by lemcoder** [CommunityToolkit/Maui#1024] [Q&A]
Is there a preffered way of using community toolkit in a MauiClassLibrary project? I have a module with 3 views and I want to use community toolkit, but I keep getting errors regarding the use of xml namespace. I've found a workaround that if I make a mock extension method and add UseMauiCommunityToolkit line - everything works fine:
```
public static MauiAppBuilder Workaround(this MauiAppBuilder builder)
        {
            builder.UseMauiCommunityToolkit();

            return builder;
        }
```

It works but I would much prefer not to add useless code inside my module...

**Comment by VladislavAntonyuk** [comment]
adding builder.UseMauiCommunityToolkit(); is required. Alternatively, you can use xmlns with explicit class and assembly

**Comment by TheCodeTraveler** [comment]
The root cause is the Linker is removing our library because it thinks you are not using it. The Linker scans C# code looking for references to `CommunityToolkit.Maui`, and if it doesn't find any, it will remove the DLL because it assumes you are not using it; it doesn't know how to scan XAML.

To ensure `CommunityToolkit.Maui` is not removed by the Linker, you should reference the library in C#; something like this:

```cs
var behavior = new EventToCommandBehavior();
```

**Comment by CliffAgius** [comment]
Closed as answered.
